Automattic/wp-calypso
# Editor: closing editor always returns to My Home after creating a new site

Issue described in #45011. Bug was introduced in #44724.

Gutenboarding doesn't have access to the Calypso store, so as you navigate through the onboarding flow we're not remembering routes. This causes the bug with the editor close button because it relies on us remembering where the user came from. This fix wipes the last non-editor route from the redux store so that users who create additional sites get the same experience as users their first site.

#### Changes proposed in this Pull Request

* Set a `a8c-clearLastNonEditorRoute` flag after completing gutenboarding
* iframe'd editor clears the `lastNonEditorRoute` redux state if the `a8c-clearLastNonEditorRoute` is present

#### Testing instructions



* Visit `/themes/{{site}}` while logged in
* Now go to `/new`
* Go through the onboarding process until you land inthe block editor
* Open the nav sidebar, the button to go back to Calypso should say My Home and take you to the customer home
* Go through the same process except purchase a plan during onboarding, to test that the fix also works if you go through the checkout before getting to the editor

Fixes #45011
